apps/console/src/features/applications/utils/sign-in-method-utils.ts

```diff
@@ -317,6 +317,86 @@ export class SignInMethodUtils {
                 idpList: []
             };
         }
+    }
+
+    public static isFederatedConflictWithSMSOTP(
+        args: FederatedConflictWithSMSOTPArgs
+    ): FederatedConflictWithSMSOTPReturnValue {
+
+        const { federatedAuthenticators, steps, subjectStepId } = args;
+
+        /**
+         * We are solving two problems:
+         *
+         * 1) Find out the subject step all federated IdPs:
+         *    We have an array of `federatedAuthenticators` and
+         *    the configured steps. We are only focused on federated
+         *    authentications configured on subject identifier step.
+         *
+         * 2) Walk forward and check other steps (step 2 and beyond) for SMS OTP:
+         *    The reason to do this check is that, if SMS OTP is configured with 
+         *    a federated authenticator, Asgardeo should receive the user's profile
+         *    (including the mobile number) configured on the federated IdP.
+         */
+
+        try {
+
+            /** Start solving the 1st problem **/
+
+            const allOptions: AuthenticatorInterface[] = flatten(
+                steps
+                    .filter(({ id } : { id: number }) => id === subjectStepId)
+                    .map(({ options } : { options: AuthenticatorInterface[] }) => options)
+            );
+
+            /**
+             * Get the list of idps configured in the subject identifier step.
+             */
+            const idPsInSubjectIdStep: GenericAuthenticatorInterface[] =
+                // Extract all the IdP names.
+                [ ...(new Set((allOptions).map(({ idp } : { idp: string }) => idp))) ]
+                    // Find the authenticator model.
+                    .map((idpName: string) => federatedAuthenticators.find(
+                        ({ name } : { name: string }) => name === idpName))
+                    // Remove all the {@code undefined|null} ones please.
+                    .filter(Boolean);
```

Comment:
Shall we break this down into two steps, like extracting idp names in first step and finding the authenticator in the second step?

Reply:
@kavinda1995 WDYT?

Reply:
Resolves via [208e95f](https://github.com/wso2/identity-apps/pull/3782/commits/208e95fee0917d1cd96dd2a8a7a755adc1b553a4)